src/pytorch_lightning/core/saving.py

```diff
@@ -250,13 +260,18 @@ def _load_state(
     obj = cls(**_cls_kwargs)
 
     # give model a chance to load something
-    obj.on_load_checkpoint(checkpoint)
+    if not isinstance(obj, ModelIO):
+        obj.on_load_checkpoint(checkpoint)
 
     if isinstance(obj, pl.LightningDataModule):
         return obj
 
     # load the state_dict on the model automatically
-    keys = obj.load_state_dict(checkpoint["state_dict"], strict=strict)
```

Comment:
I think you did this to handle the case strict=None. But I don't see that ever happening. I think the types are wrong and we should change them from `Optional[bool]` to just bool with default True, just like in the public `load_from_checkpoint` function where it is like that.

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/99643a0ed8ad27dd5bc94a5eaf7070523223b5dd